stader-cli/node/deposit-sd.go

```diff
@@ -46,63 +46,24 @@ func nodeDepositSd(c *cli.Context) error {
 		return err
 	}
 
-	// Calculate max uint256 value
-	maxApproval := big.NewInt(2)
-	maxApproval = maxApproval.Exp(maxApproval, big.NewInt(256), nil)
-	maxApproval = maxApproval.Sub(maxApproval, big.NewInt(1))
-
-	if allowance.Allowance.Cmp(maxApproval) < 0 {
-		fmt.Println("Before depositing SD, you must first give the collateral contract approval to interact with your SD.")
-		fmt.Println("This only needs to be done once for your node.")
-
-		// If a custom nonce is set, print the multi-transaction warning
-		if c.GlobalUint64("nonce") != 0 {
-			cliutils.PrintMultiTransactionNonceWarning()
-		}
-
-		// Get approval gas
-		approvalGas, err := staderClient.NodeDepositSdApprovalGas(maxApproval)
+	if allowance.Allowance.Cmp(amountWei) < 0 {
+		fmt.Println("Before depositing SD, you must first give the collateral contract approval to interact with your SD. Amount to approve: ", amountWei)
+		err = nodeApproveSdWithAmount(c, staderClient, amountWei)
```

Comment:
but now they will have to approve each time right? compared to approving the max amount?